conanfile.py

```diff
@@ -114,142 +119,239 @@ def requirements(self):
+            copy(self, pattern='*.so', dst=os.path.join(self.package_folder, "lib"), src=os.path.join(self._build_subfolder, 'rabbitmq'), keep_path=False)
         if self.options.with_clickhouse:
-            self.copy(
-                pattern='*',
-                dst='include',
-                src='clickhouse/include',
-                keep_path=True,
-            )
+            copy(self, pattern='*', dst=os.path.join(self.package_folder, "include"), src=os.path.join(self.source_folder, "clickhouse", "include"), keep_path=True)
+            copy(self, pattern='*.a', dst=os.path.join(self.package_folder, "lib"), src=os.path.join(self._build_subfolder,'clickhouse'), keep_path=False)
+            copy(self, pattern='*.so', dst=os.path.join(self.package_folder, "lib"), src=os.path.join(self._build_subfolder, 'clickhouse'), keep_path=False)
+
+        copy(self, pattern='*.a', dst=os.path.join(self.package_folder, "lib"), src=os.path.join(self._build_subfolder,'core'), keep_path=False)
+        copy(self, pattern='*.so', dst=os.path.join(self.package_folder, "lib"), src=os.path.join(self._build_subfolder, 'core'), keep_path=False)
+
+    @property
+    def _userver_components(self):
+
+        def ares():
+            return ["c-ares::c-ares"]
+        def fmt():
+            return ["fmt::fmt"]
+        def curl():
+            return ["libcurl::libcurl"]
+        def cryptopp():
+            return ["cryptopp::cryptopp"]
+        def cctz():
+            return ["cctz::cctz"]
+        def boost():
+            return ["boost::boost"]
+        def concurrentqueue():
+            return ["concurrentqueue::concurrentqueue"]
+        def yaml():
+            return ["yaml-cpp::yaml-cpp"]
+        def libev():
+            return ["libev::libev"]
+        def http_parser():
+            return ["http_parser::http_parser"]    
+        def jemalloc():
+            return ["jemalloc::jemalloc"] if self.options.with_jemalloc else []
+        def grpc():
+            return ["grpc::grpc"] if self.options.with_grpc else []
+        def postgresql():
+            return ["libpq::pq"] if self.options.with_postgresql else []
+        def gtest():
+            return ["gtest::gtest"] if self.options.with_utest else []
+        def benchmark():
+            return ["benchmark::benchmark"] if self.options.with_utest else []            
+        def mongo():
+            return ["mongo-c-driver::mongo-c-driver"] if self.options.with_mongodb else []                                    
+        def hiredis():
+            return ["hiredis::hiredis"] if self.options.with_redis else []
+        def amqpcpp():
+            return ["amqp-cpp::amqp-cpp"] if self.options.with_rabbitmq else []  
+
+        userver_components = [
+            {"target": "userver-core-internal",       "lib": "core-internal",       "requires": [] },
```

Comment:
Can we avoid publishing `userver-core-internal`? The user should not be able to link with it directly